google/gcloud/command.go

```diff
@@ -173,45 +212,56 @@ func (c *Command) authConfigureDocker(ctx context.Context, r *readline.Readline)
 }
 
 func (c *Command) containerClustersGetCredentials(ctx context.Context, r *readline.Readline) error {
-	environment := r.Args().At(0)
-	clusterName := r.Args().At(2)
-
-	clusters := c.gcloud.cfg.ClusterNamesForEnv(environment)
-	if clusterName != "" {
-		clusters = []string{clusterName}
+	var env []string
+	environment, err := c.gcloud.cfg.Environment(r.Args().At(1))
+	if err != nil {
+		return errors.Errorf("failed to retrieve environment for: %s", r.Args().At(1))
 	}
 
-	for _, cluster := range clusters {
-		serviceAccounts, err := c.gcloud.FindAccounts(ctx, environment, cluster)
-		if err != nil {
-			return err
-		}
+	cluster, err := environment.Cluster(r.Args().At(2))
+	if err != nil {
+		return errors.Errorf("failed to retrieve cluster for: %s", r.Args().At(2))
+	}
 
-		if len(serviceAccounts) > 1 {
-			c.l.Warnf("multiple accounts found for env %q and cluster %q", environment, cluster)
-		}
-		accountPath, _ := filepath.Abs(serviceAccounts[0].Path)
+	role := ClusterRoleDefault
+	if r.Args().HasIndex(3) {
+		role = r.Args().At(3)
+	}
 
-		kubectlCluster := c.kubectl.Cluster(environment + "-" + cluster)
-		gcloudCluster, ok := c.gcloud.cfg.FindCluster(environment, cluster)
-		if !ok {
-			return fmt.Errorf("could not find configuration for env %q and cluster %q", environment, cluster)
+	// resolve or retrieve service account access token
+	accessTokenFilename := path.Join(
+		os.Getenv(env2.ProjectRoot),
```

Comment:
it's a weird package name! Wouldn't it be better to change the env variable?

Reply:
that was my code completion ... gone anyway